packages/select/src/common/listItemsProps.ts

```diff
@@ -124,6 +124,22 @@ export interface IListItemsProps<T> extends IProps {
      */
     onQueryChange?: (query: string, event?: React.ChangeEvent<HTMLInputElement>) => void;
 
+    /**
+     * If provided, allows new items to be created with the current query string.
+     * This is invoked when user interaction causes a new item to be created, either by pressing the `enter` key or
+     * by clicking on the "Create Item" option. It transforms a query string into an item type.
+     */
+    createItemFromQuery?: (query: string) => T;
+
+    /**
+     * Custom renderer to transform the current query string into a selectable "Create Item" option.
+     */
+    createItemRenderer?: (
+        query: string,
+        active: boolean,
```

Comment:
when is this `active` param useful?

Reply:
This is for when the "create" item should be highlighted; for example, if you interact with the dropdown with the up/down key, when your selection is on the create item, `active` would be `true`, and the item should be rendered accordingly

Reply:
what else could you select in this case? there would be no other results to select, right? so it should be automatically "active"? I think the param is probably required for full customization of rendering, I'm not rejecting it, but maybe you could post a gif of that interaction?

Reply:
The create item is actually always rendered as long as there's input, so the "active" flag is useful:

![active-create](https://user-images.githubusercontent.com/4718844/53114099-108e7b00-3511-11e9-8fd8-653d30438af0.gif)

Reply:
And to the point of "full customization of rendering" - the consumers provide code for rendering the "create" item just like they provide code for rendering other items, so to some extent this rendering is always custom

Reply:
Should this be `isActive`, or is `active` the norm (like `disabled`)?